tidb: add tidb_analyze_column_options (#18281)
pingcap/docs: system-variables.md

- Default value: `OFF`
- This variable is used to set whether the `AUTO_INCREMENT` property of a column is allowed to be removed by executing `ALTER TABLE MODIFY` or `ALTER TABLE CHANGE` statements. It is not allowed by default.

### tidb_analyze_column_options <span class="version-mark">New in v8.3.0</span>

> **Note:**
>
> - This variable only works when [`tidb_analyze_version`](#tidb_analyze_version-new-in-v510) is set to `2`.
> - If you upgrade your TiDB cluster from a version earlier than v8.3.0 to v8.3.0 or later, this variable is set to `ALL` by default to keep the original behavior.
> - Starting from v8.3.0, for a newly deployed TiDB cluster, this variable is set to `PREDICATE` by default.

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Enumeration
- Default value: `PREDICATE`
- Value options:`ALL`, `PREDICATE`

> - This variable controls the behavior of the `ANALYZE TABLE` statement. Setting it to `PREDICATE` means only collecting statistics for [predicate columns](/statistics.md#collect-statistics-on-some-columns); setting it to `ALL` means collecting statistics for all columns. In scenarios where OLAP queries are used, it is recommended to set it to `ALL`, otherwise query performance can significantly degrade.

Reviewer: ```suggestion
- This variable controls the behavior of the `ANALYZE TABLE` statement. Setting it to `PREDICATE` means only collecting statistics for [predicate columns](/statistics.md#collect-statistics-on-some-columns); setting it to `ALL` means collecting statistics for all columns. In scenarios where OLAP queries are used, it is recommended to set it to `ALL`. Otherwise collecting statistics can result in a significant drop in query performance.
```

- Default value: `OFF`
- This variable is used to set whether the `AUTO_INCREMENT` property of a column is allowed to be removed by executing `ALTER TABLE MODIFY` or `ALTER TABLE CHANGE` statements. It is not allowed by default.

### tidb_analyze_column_options <span class="version-mark">New in v8.3.0</span>

> **Note:**
>
> This variable only works when [`tidb_analyze_version`](#tidb_analyze_version-new-in-v510) is set to `2`.
>

> > After upgrading a TiDB cluster from a version earlier than v8.3.0, this variable be set to `ALL` to keep the original behavior.

Reviewer 2: ```suggestion
> If you upgrade your TiDB cluster from a version earlier than v8.3.0 to v8.3.0 or later, this variable is set to `ALL` by default to keep the original behavior.
```

>

> > Since v8.3.0, for a newly deployed TiDB cluster, this variable is set to `PREDICATE` by default.

Reviewer 2: The word "since" is usually used with the present perfect tense.
```suggestion
> Starting from v8.3.0, for a newly deployed TiDB cluster, this variable is set to `PREDICATE` by default.
```

> > This variable only works when [`tidb_analyze_version`](#tidb_analyze_version-new-in-v510) is set to `2`.
> >
> > If you upgrade your TiDB cluster from a version earlier than v8.3.0 to v8.3.0 or later, this variable is set to `ALL` by default to keep the original behavior.
> >
> > Starting from v8.3.0, for a newly deployed TiDB cluster, this variable is set to `PREDICATE` by default.

Reviewer 2: ```suggestion
> - This variable only works when [`tidb_analyze_version`](#tidb_analyze_version-new-in-v510) is set to `2`.
> - If you upgrade your TiDB cluster from a version earlier than v8.3.0 to v8.3.0 or later, this variable is set to `ALL` by default to keep the original behavior.
> - Starting from v8.3.0, for a newly deployed TiDB cluster, this variable is set to `PREDICATE` by default.
```

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Enumeration
- Default value: `PREDICATE`
- Value options:`ALL`, `PREDICATE`

> - This variable controls the behavior of the ANALYZE TABLE statement. Setting it to PREDICATE means only collecting statistics for [predicate columns](/statistics.md#collect-statistics-on-some-columns) while setting it to ALL means collecting statistics for all columns. In scenarios where `OLAP` queries are used, it is recommended to set it to `ALL`, otherwise query performance will significantly degrade.

Reviewer 2: ```suggestion
- This variable controls the behavior of the `ANALYZE TABLE` statement. Setting it to `PREDICATE` means only collecting statistics for [predicate columns](/statistics.md#collect-statistics-on-some-columns); setting it to `ALL` means collecting statistics for all columns. In scenarios where OLAP queries are used, it is recommended to set it to `ALL`, otherwise query performance can significantly degrade.
```

- Default value: `ON`
- This variable controls whether to record the execution information of each operator in the slow query log and whether to record the [usage statistics of indexes](/information-schema/information-schema-tidb-index-usage.md).

### tidb_analyze_column_options <span class="version-mark">New in v8.3.0</span>

> **Note:**
>
> This variable only works when [`tidb_analyze_version`](#tidb_analyze_version-new-in-v510) is set to `2`.
>
> After upgrading a TiDB cluster from a version earlier than v8.3.0, this variable be set to `ALL` to keep the original behavior.
>
> Since v8.3.0, for a newly deployed TiDB cluster, this variable is set to `PREDICATE` by default.

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Enumeration
- Default value: `PREDICATE`
- Value options:`ALL`, `PREDICATE`

> - This variable controls the behavior of the ANALYZE TABLE statement. Setting it to PREDICATE means only collecting statistics for predicate columns, while setting it to ALL means collecting statistics for all columns. In scenarios where `OLAP` queries are used, it is recommended to set it to `ALL`, otherwise query performance will significantly degrade.

Reviewer 3: Should we also explain the meaning of "predicate columns"? For exmaple, columns that appear in Range or Eq predicates.

> **Warning:**
>

> > Starting from v8.3.0, this variable is deprecated. TiDB tracks predicate columns by default.

Reviewer 2: ```suggestion
> Starting from v8.3.0, this variable is deprecated. TiDB tracks predicate columns by default. For more information, see [`tidb_analyze_column_options`](#tidb_analyze_column_options-new-in-v830).
```
